integration-test/src/test/java/com/datastrato/gravitino/integration/test/catalog/jdbc/mysql/CatalogMysqlIT.java

```diff
@@ -1225,4 +1225,98 @@ void testMySQLTableNameCaseSensitive() {
     assertionsTableInfo(
         "TABLENAME", table_comment, Arrays.asList(newColumns), properties, indexes, table);
   }
+
+  @Test
+  void testMySQLSchemaNameCaseSensitive() {
+    Column col1 = Column.of("col_1", Types.LongType.get(), "id", false, false, null);
+    Column col2 = Column.of("col_2", Types.VarCharType.of(255), "code", false, false, null);
+    Column col3 = Column.of("col_3", Types.VarCharType.of(255), "config", false, false, null);
+    Column[] newColumns = new Column[] {col1, col2, col3};
+
+    Index[] indexes = new Index[] {Indexes.unique("u1_key", new String[][] {{"col_2"}, {"col_3"}})};
+
+    String schema_1 = "db_";
+    String schema_2 = "db_1";
+    String schema_3 = "db_2";
+    String schema_4 = "db12";
+    SupportsSchemas schemas = catalog.asSchemas();
+    schemas.createSchema(
+        NameIdentifier.of(metalakeName, catalogName, schema_1), null, Collections.emptyMap());
+    schemas.createSchema(
+        NameIdentifier.of(metalakeName, catalogName, schema_2), null, Collections.emptyMap());
+    schemas.createSchema(
+        NameIdentifier.of(metalakeName, catalogName, schema_3), null, Collections.emptyMap());
+    schemas.createSchema(
+        NameIdentifier.of(metalakeName, catalogName, schema_4), null, Collections.emptyMap());
+
+    NameIdentifier[] nameIdentifiers = schemas.listSchemas(Namespace.of(metalakeName, catalogName));
```

Comment:
please use `load` a schema here, you will find the logic `load` and `list` is different in the case of using JDBC metadata.